### PR TITLE
Use type mapper in Tables

### PR DESCRIPTION
**Pull request by pakrym**
I haven't profiled this so there might be more low-hanging fruit.

# Before

```
|      Method |     Mean |     Error |    StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------ |---------:|----------:|----------:|-------:|------:|------:|----------:|
|   Serialize | 4.006 us | 0.0799 us | 0.2106 us | 0.2747 |     - |     - |    1792 B |
| Deserialize | 5.553 us | 0.1030 us | 0.0860 us | 0.1450 |     - |     - |     952 B |
```

# After
```


|      Method |     Mean |     Error |    StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------ |---------:|----------:|----------:|-------:|------:|------:|----------:|
|   Serialize | 1.832 us | 0.0364 us | 0.0753 us | 0.2632 |     - |     - |    1696 B |
| Deserialize | 3.171 us | 0.0637 us | 0.0759 us | 0.0267 |     - |     - |     200 B |

```

TODO: the TypeBinder supports more features than Tables does (DataMember attribute, IgnoreAttribute) what should we do with these?

### CODE REVIEW

File: common/Perf/Azure.Test.Perf/PerfProgram.cs [hunk 1]
```diff
@@ -31,6 +32,13 @@ public static class PerfProgram
 
         public static async Task Main(Assembly assembly, string[] args)
         {
+            // See if we want to run a BenchmarkDotNet microbenchmark
+            if (args.Length > 0 && args[0].Equals("micro", StringComparison.OrdinalIgnoreCase))
```
Comment (pakrym) [inline common/Perf/Azure.Test.Perf/PerfProgram.cs:36]: This is the change to allow running BDN tests through the Perf runner using the `dotnet run -- micro ..` command line. 

TODO: discuss with the team if this is the way we want it.

cc @mikeharder

**Review by christothes — APPROVED** [review]

File: common/Perf/Azure.Test.Perf/PerfProgram.cs (hunk 1 above)
Comment (mikeharder) [reply common/Perf/Azure.Test.Perf/PerfProgram.cs:36]: This change looks fine to me.

File: sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs [hunk 2]
```diff
@@ -117,10 +115,11 @@ internal static void CastAndRemoveAnnotations(this IDictionary<string, object> e
         internal static List<T> ToTableEntityList<T>(this IReadOnlyList<IDictionary<string, object>> entityList) where T : class, ITableEntity, new()
         {
             var result = new List<T>(entityList.Count);
+            var typeInfo = TablesTypeBinder.Shared.GetBinderInfo(typeof(T));
```
Comment (christothes) [inline sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs:118]: Would it make sense to cache this?

**Review by christothes — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs (hunk 2 above)
Comment (pakrym) [reply sdk/tables/Azure.Data.Tables/src/Extensions/DictionaryTableExtensions.cs:118]: GetBinderInfo cached internally.

**Pull request merged**